addons/binding/org.openhab.binding.elerotransmitterstick/src/main/java/org/openhab/binding/elerotransmitterstick/discovery/EleroChannelDiscoveryService.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.elerotransmitterstick.discovery;
+
+import static org.openhab.binding.elerotransmitterstick.EleroTransmitterStickBindingConstants.*;
+
+import java.util.Collections;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.elerotransmitterstick.handler.EleroTransmitterStickHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EleroChannelDiscoveryService} is responsible for discovery of elero channels from an Elero Transmitter
+ * Stick.
+ *
+ * @author Volker Bier - Initial contribution
+ */
+public class EleroChannelDiscoveryService extends AbstractDiscoveryService {
+    private static final int DISCOVER_TIMEOUT_SECONDS = 30;
+    private final Logger logger = LoggerFactory.getLogger(EleroChannelDiscoveryService.class);
+
+    EleroTransmitterStickHandler bridge;
+
+    /**
+     * Creates the discovery service for the given handler and converter.
+     */
+    public EleroChannelDiscoveryService(EleroTransmitterStickHandler stickHandler) {
+        super(Collections.singleton(THING_TYPE_ELERO_CHANNEL), DISCOVER_TIMEOUT_SECONDS, true);
+
+        bridge = stickHandler;
+    }
+
+    @Override
+    protected void startScan() {
+        discoverSensors();
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        discoverSensors();
+    }
+
+    private void discoverSensors() {
+        if (bridge.getStick() == null) {
+            logger.debug("Stick not opened, scanning postponed.");
+            return;
+        }
+
+        int[] channelIds = null;
```

Comment:
Can you please use an `ArrayList` since we are in Java, not in C ;)

Reply:
I thought primitive arrays are part of the language specification so people can use them. Is see no reason to use an Arraylist, as the size is restricted by the hardware and will not change. The ArrayList only adds memory overhead and unneeded functionality.

Reply:
For sure arrays are part of the language specification, but it is a bad style to use array indices as an identifier for an object. It is a lot better to add this identifier to a channel object and store these in an `ArrayList`. Regarding the memory footprint... As long as you do not store tons of megabyte we do not have to worry, we are not running on arduino nanos ;)